app/packages/jaeger/src/utils/utils.ts

```diff
@@ -0,0 +1,364 @@
+import { getChartColor } from '@kobsio/core';
+
+import TreeNode from './TreeNode';
+
+export const description =
+  'Open source, end-to-end distributed tracing: Monitor and troubleshoot transactions in complex distributed systems.';
+
+export const spanKinds = ['unspecified', 'internal', 'server', 'client', 'producer', 'consumer'];
+
+/**
+ * The following interfaces are used to represent a trace like it returned from the API.
+ * See: https://github.com/jaegertracing/jaeger-ui/blob/master/packages/jaeger-ui/src/types/trace.tsx
+ */
+export interface IKeyValuePair {
+  key: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  value: any;
+}
+
+export interface ILog {
+  fields: IKeyValuePair[];
+  timestamp: number;
+}
+
+export interface IProcess {
+  serviceName: string;
+  tags: IKeyValuePair[];
+}
+
+export interface ISpanReference {
+  refType: 'CHILD_OF' | 'FOLLOWS_FROM';
+  // eslint-disable-next-line no-use-before-define
+  span: ISpan | null | undefined;
+  spanID: string;
+  traceID: string;
+}
+
+export interface ISpanData {
+  duration: number;
+  logs: ILog[];
+  operationName: string;
+  processID: string;
+  references?: ISpanReference[];
+  spanID: string;
+  startTime: number;
+  tags?: IKeyValuePair[];
+  traceID: string;
+  warnings?: string[] | null;
+}
+
+export interface ISpan extends ISpanData {
+  depth: number;
+  hasChildren: boolean;
+  process: IProcess;
+  references: NonNullable<ISpanData['references']>;
+  relativeStartTime: number;
+  subsidiarilyReferencedBy: ISpanReference[];
+  tags: NonNullable<ISpanData['tags']>;
+  warnings: NonNullable<ISpanData['warnings']>;
+}
+
+export interface ITraceData {
+  processes: Record<string, IProcess>;
+  traceID: string;
+}
+
+export interface ITrace extends ITraceData {
+  duration: number;
+  endTime: number;
+  services: { name: string; numberOfSpans: number }[];
+  spans: ISpan[];
+  startTime: number;
+  traceName: string;
+}
+
+/**
+ * `IDeduplicateTags` is the interface, which is returned by the `deduplicateTags` function.
+ */
+export interface IDeduplicateTags {
+  tags: IKeyValuePair[];
+  warnings: string[];
+}
+
+// formatTraceTime is a wrapper around the formatTime function from the core plugin.
+export const formatTraceTime = (timestamp: number): string => {
+  const d = new Date(Math.floor(timestamp / 1000));
+
+  return `${d.getFullYear()}-${('0' + (d.getMonth() + 1)).slice(-2)}-${('0' + d.getDate()).slice(-2)} ${(
+    '0' + d.getHours()
+  ).slice(-2)}:${('0' + d.getMinutes()).slice(-2)}:${('0' + d.getSeconds()).slice(-2)}.${(
+    '0' + d.getMilliseconds()
+  ).slice(-3)}`;
```

Comment:
I'd prefer a date library to do this, instead of doing this by hand, e.g:
https://day.js.org/docs/en/display/format

Reply:
I do not really see the benefit of adding another dependency for this. We only have this in two places for now and currently there are now plans to support other formats then these ones.